### PR TITLE
DataFlash: log dataflash-file statistics periodically

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by OXINARF** [comment]
@peterbarker Why do we keep a counter of internal errors? It doesn't look like we use it for something? Maybe I missed it.

File: libraries/DataFlash/DataFlash_File.cpp [hunk 1]
```diff
@@ -197,6 +199,7 @@ void DataFlash_File::periodic_1Hz(const uint32_t now)
         _write_fd = -1;
         _initialised = false;
     }
+    df_stats_log();
```
Comment (OXINARF) [inline libraries/DataFlash/DataFlash_File.cpp]: Should this be inside an else? I mean, what is the point of writing the stats if the IO thread is dead?

**Comment by peterbarker** [comment]
OK, underscores returned :-)

File: libraries/DataFlash/DataFlash_File.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/DataFlash/DataFlash_File.cpp]: Still nice to know what's going on.  It's not going to hurt anybody and is slightly easier to read (IMHO)

Comment (OXINARF) [reply libraries/DataFlash/DataFlash_File.cpp]: I'm not sure how you will know what's going on if it isn't written into the log, but I'm not fussed about this.

**Comment by OXINARF** [comment]
@peterbarker They didn't return in this PR 🙂 Can you rebase this on top of #6826?

**Comment by peterbarker** [comment]
On Sat, 26 Aug 2017, Francisco Ferreira wrote:

> @peterbarker They didn't return in this PR 🙂 Can you rebase this on top of
> #6826?

I was waiting to rebase it on master :-P

Rebased it on 6826 instead.

**Comment by OXINARF** [comment]
Merged, thanks!

**Pull request merged**